commaai/openpilot
# Enhance SConstruct by enabling arbitrary compiler flags

**Description** 

The commit introduces a new option for Sconstruct to allow users to pass arbitrary compiler flags via the command line.

The new design introduces greater flexibility for the build process allowing the user to pass any compiler flag(s) as needed. This is achieved by using the `--ccflags` option followed by the desired flags.

For example: 
```sh
scons -u -j$(nproc) --ccflags="-fno-inline -O3 -Wall"
```

This command would add `-fno-inline`, `-O3`, and `-Wall` to `ccflags`.

**Verification** 

1. Try building with different flags using the `--ccflags` option.
2. Check that the passed flags are correctly used in the build process.
